lib/usd/utils/MergePrimsOptions.h

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsdUtils/Api.h>
+
+#include <pxr/base/tf/staticTokens.h>
+#include <pxr/base/tf/token.h>
+#include <pxr/base/vt/dictionary.h>
+
+namespace MayaUsdUtils {
+
+//----------------------------------------------------------------------------------------------------------------------
+/// MergeVerbosity level flags.
+
+enum class MergeVerbosity
+{
+    None = 0,
+    Same = 1 << 0,
+    Differ = 1 << 1,
+    Child = 1 << 2,
+    Children = 1 << 3,
+    Failure = 1 << 4,
+    Default = Differ | Child | Children | Failure,
+};
+
+inline MergeVerbosity operator|(MergeVerbosity a, MergeVerbosity b)
+{
+    return MergeVerbosity(uint32_t(a) | uint32_t(b));
+}
+
+inline MergeVerbosity operator&(MergeVerbosity a, MergeVerbosity b)
+{
+    return MergeVerbosity(uint32_t(a) & uint32_t(b));
+}
+
+inline MergeVerbosity operator^(MergeVerbosity a, MergeVerbosity b)
+{
+    return MergeVerbosity(uint32_t(a) ^ uint32_t(b));
+}
+
+inline bool contains(MergeVerbosity a, MergeVerbosity b) { return (a & b) != MergeVerbosity::None; }
+
+//----------------------------------------------------------------------------------------------------------------------
+// Missing field handling flags.
+
+enum class MergeMissing
+{
+    None = 0,
+
+    // If set, attributes found only in the source are created in the destination.
+    Create = 1 << 0,
+
+    // If set, attributes missing from the source are preserved in the destination.
+    Preserve = 1 << 1,
+
+    All = Create | Preserve,
+};
+
+inline MergeMissing operator|(MergeMissing a, MergeMissing b)
+{
+    return MergeMissing(uint32_t(a) | uint32_t(b));
+}
+
+inline MergeMissing operator&(MergeMissing a, MergeMissing b)
+{
+    return MergeMissing(uint32_t(a) & uint32_t(b));
+}
+
+inline MergeMissing operator^(MergeMissing a, MergeMissing b)
+{
+    return MergeMissing(uint32_t(a) ^ uint32_t(b));
+}
+
+inline bool contains(MergeMissing a, MergeMissing b) { return (a & b) != MergeMissing::None; }
+
+//----------------------------------------------------------------------------------------------------------------------
+// Options to control prims merging.
+
+struct MergePrimsOptions
+{
+    // How much logging is done during the merge.
+    MergeVerbosity verbosity = MergeVerbosity::Default;
```

Comment:
Nit-pick: we should be using `{}` for initialization.  See
https://github.com/Autodesk/maya-usd/blob/dev/doc/codingGuidelines.md#modern-c

Reply:
Do enum have a default value? I don't see how it would know that Default is the default value.

Reply:
Just change to
MergeVerbosity verbosity{MergeVerbosity::Default};

Reply:
Ah, OK, I misunderstood the suggestion!